### PR TITLE
Wsc 016 adding pressure sensor

### LINKED ISSUE #34: Add barometric pressure sensor BMP280
Add [BMP280](https://www.reichelt.de/de/de/entwicklerboards-temperatur-und-drucksensor-bmp280-debo-bmp280-p266034.html?PROVID=2788&gclid=CjwKCAjwu5CDBhB9EiwA0w6sLWpL1WYVLSI4FrHcvoOxq-G1cc20At6Bncs8owHdyAhnD-hVDgVOExoCaXUQAvD_BwE&&r=1)  sensor to the board

Create a basic interface that can be 'blended' with current code base

Publish the pressure synchronously or asynchronously with temperature on a terminal/console

### PR DESCRIPTION
**Pull request by ggulgulia**
* Publishing pressure data with BMP280 sensor
* Improved message on the client side

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,5 +1,4 @@
 build
 Dockerfile
 Image
-.git
```
Comment (Anokhi1994) [inline .dockerignore:4]: hmm... fishy.

File: sensor_interface/CMakeLists.txt [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
 
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wpedantic -Werror -Wextra")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
```
Comment (Anokhi1994) [inline sensor_interface/CMakeLists.txt:11]: hmmm... more fishy.

File: sensor_interface/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    settings = "cppstd", "os", "compiler", "build_type", "arch"
+    requires = "paho-mqtt-cpp/1.2.0", "paho-mqtt-c/1.3.8", "mosquitto/2.0.7"
+    options = {"shared": [True, False]}
+    default_options = {"shared": False}
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
```
Comment (Anokhi1994) [inline sensor_interface/conanfile.py:16]: very good work here!

File: sensor_interface/headers/PressurePublisher.hpp [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+#include <exception>
+namespace WS {
+
+    class PressurePublisher {
+    public:
+        PressurePublisher(std::shared_ptr<MQTTClient> client);
+        double publish() const;
+        auto get_client() -> std::shared_ptr<MQTTClient> const;
+
+    private:
```
Comment (Anokhi1994) [inline sensor_interface/headers/PressurePublisher.hpp:15]: super!

File: sensor_interface/headers/mqtt_client.hpp [hunk 5]
```diff
@@ -11,6 +11,14 @@
...
 using namespace std::literals::chrono_literals;
 
+template<typename T>
+struct is_string
+        : public std::disjunction<
+                std::is_same<char *, typename std::decay_t<T>>,
+                std::is_same<const char *, typename std::decay_t<T>>,
+                std::is_same<std::string, typename std::decay_t<T>>
+        > { };
+
```
Comment (Anokhi1994) [inline sensor_interface/headers/mqtt_client.hpp:21]: what a code!

File: sensor_interface/headers/mqtt_client.hpp [hunk 6]
```diff
@@ -31,10 +39,10 @@ namespace WS {
     Result MQTTClient::PublishMessage(Topic&& topic, Payload&& payload)
     {
         Result retval = Result::Successful;
-        if constexpr (std::is_arithmetic_v<Payload>) {
+        if constexpr (std::is_arithmetic_v<Payload> && is_string<decltype(topic)>::value) {
             try {
```
Comment (Anokhi1994) [inline sensor_interface/headers/mqtt_client.hpp:43]: Very good checking here

File: sensor_interface/src/main.cpp [hunk 7]
```diff
@@ -51,10 +54,12 @@ int main(){
     std::cout << "Hello Weather Station\n";
     std::shared_ptr<MQTTClient> client = std::make_shared<MQTTClient>("localhost", "main");
     TemperaturePublisher temp_publisher{client};
-
+    PressurePublisher pressure_publisher{client};
+    std::thread pressureThread(PressureHelper);
```
Comment (Anokhi1994) [inline sensor_interface/src/main.cpp]:
```suggestion
```

**Review by Anokhi1994 — CHANGES_REQUESTED** [review]

**Review by ggulgulia — COMMENTED** [review]
changed as requested

**Pull request merged**